### PR TITLE
Fix formatting of lib's fun starting with uppercase letter

### LINKED ISSUE #5432: Formatter does not support upper-case fun names
Since #4934 fun names can start with an uppercase letter. Compiling works fine but it seems the formatter is not aware of this change.

Example:
```crystal
# mylib.cr
lib MyLib
  fun Foo()
end
```
```
$ crystal tool format mylib.cr
Error: couldn't format 'mylib.cr', please report a bug including the contents of it: https://github.com/crystal-lang/crystal/issues

expecting IDENT, not `CONST, Foo`, at :2:7 (Exception)
```

### PR DESCRIPTION
**Pull request by bew**
Fixes #5432

This is my first PR on the formatter :tada: :smiley:

### CODE REVIEW

File: spec/compiler/formatter/formatter_spec.cr [hunk 1]
```diff
@@ -581,6 +581,8 @@ describe Crystal::Formatter do
   assert_format "lib Foo\nend"
   assert_format "lib Foo\ntype  Foo  =   Bar\nend", "lib Foo\n  type Foo = Bar\nend"
   assert_format "lib Foo\nfun foo\nend", "lib Foo\n  fun foo\nend"
+  assert_format "lib Foo\nfun Bar\nend", "lib Foo\n  fun Bar\nend"
+  assert_format "lib Foo\nfun bar = Bar\nend", "lib Foo\n  fun bar = Bar\nend"
```
Comment (RX14) [inline spec/compiler/formatter/formatter_spec.cr]: What about `lib Foo\nfun Foo = Bar\nend`?

We should probably spec calling functions with `CONST` names too to make sure that doesn't get broken.

File: src/compiler/crystal/tools/formatter.cr [hunk 2]
```diff
@@ -1530,9 +1530,13 @@ module Crystal
     def visit(node : FunDef)
       write_keyword :fun, " "
 
-      check :IDENT
-      write node.name
-      next_token_skip_space
+      case @token.type
+      when :IDENT, :CONST
```
Comment (straight-shoota) [inline src/compiler/crystal/tools/formatter.cr]: Maybe it would be better to let `check` receive multiple arguments and check for any of them?

Comment (bew) [reply src/compiler/crystal/tools/formatter.cr]: was thinking about it too, will probably do that

Comment (RX14) [reply src/compiler/crystal/tools/formatter.cr]: Is this needed anywhere inside `formatter.cr`? If not, put it in this PR, if it *is* used in other places, do what those places do and refactor in a new PR.

File: spec/compiler/formatter/formatter_spec.cr (hunk 1 above)
Comment (bew) [reply spec/compiler/formatter/formatter_spec.cr]: about calling, do you mean add a formatter spec for the lib-uppercase-fun calling?

File: src/compiler/crystal/tools/formatter.cr [hunk 3]
```diff
@@ -4577,8 +4573,8 @@ module Crystal
       raise "expecting keyword #{keywords.join " or "}, not `#{@token.type}, #{@token.value}`, at #{@token.location}" unless keywords.any? { |k| @token.keyword?(k) }
     end
 
-    def check(token_type)
-      raise "expecting #{token_type}, not `#{@token.type}, #{@token.value}`, at #{@token.location}" unless @token.type == token_type
+    def check(*token_types)
+      raise "expecting #{token_types.join " or "}, not `#{@token.type}, #{@token.value}`, at #{@token.location}" unless token_types.includes? @token.type
```
Comment (oprypin) [inline src/compiler/crystal/tools/formatter.cr]: Splitting this from one line is overdue now 😄

Comment (RX14) [reply src/compiler/crystal/tools/formatter.cr]: Just change the postfix `unless` to normal `unless` and it's readable enough.

File: src/compiler/crystal/tools/formatter.cr (hunk 2 above)
Comment (bew) [reply src/compiler/crystal/tools/formatter.cr]: Good idea, it looks like it was the only place where a `case @token.type` was used this way. So I'll leave this change in this PR.

File: spec/compiler/formatter/formatter_spec.cr (hunk 1 above)
Comment (RX14) [reply spec/compiler/formatter/formatter_spec.cr]: @bew yes

Comment (bew) [reply spec/compiler/formatter/formatter_spec.cr]: done!

**Comment by bew** [comment]
I squashed so it should be merge-able as is

**Comment by straight-shoota** [comment]
Great first formatter contribution! =)

**Comment by bew** [comment]
@straight-shoota thanks! I'm glad to have indirectly helped the windows stuff going on these days, by unblocking your pr :smile:

**Pull request merged**